### PR TITLE
Cleanup various repository documentation files

### LINKED ISSUE #82495: CONTRIBUTING.md link to editor translations readme is broken
### Godot version

4.1.1-stable

### System information

Windows 11

### Issue description

When clicking "translation readme" in the "CONTRIBUTING.md" file:
image

It shows:
Screenshot 2023-09-28 151721

 This means important information for new translators wanting to contribute is missing.

I believe, this issue is cause by not having the proper link attached. The current link seems to taken from an old file that is no longer present in the current version of the project (editor/translations/README.md).

I think the intended link is:
https://docs.godotengine.org/en/latest/contributing/documentation/editor_and_docs_localization.html
My reasoning is because the information in this link seems to be intended to be read by new translators but can only be found by diving deeper into other links, making the instructions confusing.

Perhaps, a possible solution would be to simply change the link to the one I provided in the CONTRIBUTING.md file, or just removing it entirely until a new README file is created.

### Steps to reproduce

In the "CONTRIBUTING.md" file go to the headline that says "Contributing to Godot's translation" and click on "translation readme"

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by akien-mga**
- Reformat logo license as a plain text file.
- Fix outdated links or references to SFC or Visual Script.
- Tweak contents of `CONTRIBUTING.md` to highlight contributor docs more prominently, and make it easier to parse.
- Tweak formatting and contents in `thirdparty/README.md` for consistency.

---

- Fixes #82495.
- Supersedes #82989.
- Supersedes #83073.
- Supersedes #83090.
- This docs PR should be merged alongside: https://github.com/godotengine/godot-docs/pull/8215

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -142,23 +122,59 @@ By fixing the regulation system via an added binding to the internal feature,
 this commit now ensures that Godot will not go past the ebullition temperature
 of cooking oil under normal atmospheric conditions.
 
-Fixes #1789, long live the Realm!
+Fixes #1340.
```
[the remark this answers is not in the document]
Comment (akien-mga) [reply CONTRIBUTING.md:125]: My old joke about the French revolution was a bit weird and out of place.

**Review by YuriSizov — APPROVED** [review]
Nice!

**Comment by AThousandShips** [comment]
Can we add a line about updating to the latest version to the reporting section?

Because been quite a few bug reports from 4.1.0 for bugs solved in 4.1.1 already

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,28 +1,44 @@
...
+
+Make sure that the bug you are experiencing is reproducible in the latest Godot
+releases. You can find an overview of all Godot releases [on the website](https://godotengine.org/download/archive/)
+to confirm whether your current version is the latest one. It's worth testing
+against both the latest stable release and the latest dev snapshot for the next
+Godot release.
+
+If you run into a bug which wasn't present in an earlier Godot version (what we
+call a _regression_), please mention it and clarify which versions you tested
+(both the one(s) working and the one(s) exhibiting the bug).
```
Comment (akien-mga) [inline CONTRIBUTING.md:34]: Add these two paragraphs to address @AThousandShips' suggestion.

Comment (AThousandShips) [reply CONTRIBUTING.md:34]: Looks great!

**Review by AThousandShips — APPROVED** [review]

**Pull request merged**